servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ClientStrategyInfluencerChainBuilder.java

```diff
@@ -46,32 +58,68 @@ private ClientStrategyInfluencerChainBuilder(ClientStrategyInfluencerChainBuilde
     }
 
     void add(StreamingHttpClientFilterFactory clientFilter) {
-        clientChain = clientChain.merge(clientFilter.requiredOffloads());
+        HttpExecutionStrategy filterOffloads = clientFilter.requiredOffloads();
+        if (defaultStrategy() == filterOffloads || noOffloadsStrategy() == filterOffloads) {
+            LOGGER.warn("Ignoring illegal filter required strategy ({}) for {}", filterOffloads, clientFilter);
+            filterOffloads = anyStrategy();
+        }
+        if (defaultStrategy() != filterOffloads && filterOffloads.hasOffloads()) {
+            clientChain = null != clientChain ? clientChain.merge(filterOffloads) : filterOffloads;
+        }
     }
 
     void add(HttpLoadBalancerFactory<?> lb) {
-        clientChain = clientChain.merge(lb.requiredOffloads());
+        HttpExecutionStrategy filterOffloads = lb.requiredOffloads();
+        if (defaultStrategy() == filterOffloads || noOffloadsStrategy() == filterOffloads) {
+            LOGGER.warn("Ignoring illegal load balancer required strategy ({}) for {}", filterOffloads, lb);
+            filterOffloads = anyStrategy();
+        }
+        if (defaultStrategy() != filterOffloads && filterOffloads.hasOffloads()) {
+            clientChain = null != clientChain ? clientChain.merge(filterOffloads) : filterOffloads;
+        }
     }
 
     void add(ConnectionFactoryFilter<?, FilterableStreamingHttpConnection> connectionFactoryFilter) {
-        connFactoryChain = connFactoryChain.merge(connectionFactoryFilter.requiredOffloads());
+        ExecutionStrategy filterOffloads = connectionFactoryFilter.requiredOffloads();
+        if (defaultStrategy() == filterOffloads || noOffloadsStrategy() == filterOffloads) {
+            LOGGER.warn("Ignoring illegal connection factory required strategy ({}) for {}",
+                    filterOffloads, connectionFactoryFilter);
+            filterOffloads = anyStrategy();
+        }
+        if (defaultStrategy() != filterOffloads && filterOffloads.hasOffloads()) {
+            connFactoryChain = null != connFactoryChain ?
+                    connFactoryChain.merge(filterOffloads) : ConnectAndHttpExecutionStrategy.from(filterOffloads);
+        }
     }
 
     void add(StreamingHttpConnectionFilterFactory connectionFilter) {
-        connFilterChain = connFilterChain.merge(connectionFilter.requiredOffloads());
+        HttpExecutionStrategy filterOffloads = connectionFilter.requiredOffloads();
+        if (defaultStrategy() == filterOffloads || noOffloadsStrategy() == filterOffloads) {
```

Comment:
consider sharing code for the above methods. IIUC the code is the same except for the log statement?

Reply:
slightly improved. Differing types prevent more sharing.